src/IPv4Block.php

```diff
@@ -20,4 +20,60 @@
 class IPv4Block extends IPBlock
 {
     protected static $ip_class = IPv4::class;
+
+    const netmask2prefix = array(
+        "0.0.0.0" => 0,
+        "128.0.0.0" => 1,
+        "192.0.0.0" => 2,
+        "224.0.0.0" => 3,
+        "240.0.0.0" => 4,
+        "248.0.0.0" => 5,
+        "252.0.0.0" => 6,
+        "254.0.0.0" => 7,
+        "255.0.0.0" => 8,
+        "255.128.0.0" => 9,
+        "255.192.0.0" => 10,
+        "255.224.0.0" => 11,
+        "255.240.0.0" => 12,
+        "255.248.0.0" => 13,
+        "255.252.0.0" => 14,
+        "255.254.0.0" => 15,
+        "255.255.0.0" => 16,
+        "255.255.128.0" => 17,
+        "255.255.192.0" => 18,
+        "255.255.224.0" => 19,
+        "255.255.240.0" => 20,
+        "255.255.248.0" => 21,
+        "255.255.252.0" => 22,
+        "255.255.254.0" => 23,
+        "255.255.255.0" => 24,
+        "255.255.255.128" => 25,
+        "255.255.255.192" => 26,
+        "255.255.255.224" => 27,
+        "255.255.255.240" => 28,
+        "255.255.255.248" => 29,
+        "255.255.255.252" => 30,
+        "255.255.255.254" => 31,
+        "255.255.255.255" => 32
+        );
+
+    /**
+     * @internal
+     * Check if the prefix is valid,
+     * rewrites IPv4 old-style netmask to CIDR prefix number,
+     * and returns the prefix.
+     *
+     * @param mixed $prefix
+     *
+     * @throws \InvalidArgumentException
+     * @return int
+     */
+    protected function checkPrefix($prefix)
+    {
+        if (isset(self::netmask2prefix[$prefix])) {
+            return (int) self::netmask2prefix[$prefix];
```

Comment:
I think this whole block is a little overkill. You can accomplish the same thing with a much more elegant method...

Reply:
Based on your comment and on your commit [here](https://github.com/rlanvin/php-ip/pull/54/commits/b72e328136c4056ccb1b6862874766eab2274ce9), it looks like this is a discussion between having a bulky pre-calculated map or doing the math on the fly, every time. I'm personally more in favor of having a pre-calculated map. My reasoning when I do pros and cons of the map:
Pros: run time execution is much, much faster (checking isset) and code is simpler to read
Cons: slightly more memory usage I guess (although probably negligible?), code less elegant (somewhat subjective)

I'm in favor of faster run time, performance is really important for a low level library like this. But happy to hear everyone's opinion.

Reply:
Runtime speed was the main reason I selected this kind of ugly map instead of any real parsing. I also hoped that with the const definition it would be only once in memory no matter how many object instances you create, but I did not verify that.

Reply:
The same speed reason was to implement it in IPv4Block.php only, and not in the generic IPBlock with some if $version check, to only get any penalty for IPv4, and not for the future more and more important IPv6.

Reply:
Thank you for your input, good enough for me.